src/ionosphere/waccmx/edyn_esmf.F90

```diff
@@ -1689,13 +1606,30 @@ subroutine edyn_esmf_regrid_mag2geo(srcfield, dstfield, ndim)
         call edyn_esmf_chkerr(subname, 'ESMF_FieldRegrid geo2mag 3D', rc)
      end if
    end subroutine edyn_esmf_regrid_mag2geo
-   !-----------------------------------------------------------------------
 
+   !-----------------------------------------------------------------------
 
    subroutine edyn_esmf_update_flag( flag )
      logical, intent(in) :: flag
      edyn_esmf_update_step=flag
    end subroutine edyn_esmf_update_flag
 
+   !-----------------------------------------------------------------------
+
+   subroutine edyn_esmf_update_phys_mesh(new_phys_mesh)
+      use ESMF, only: ESMF_Mesh, ESMF_MeshIsCreated, ESMF_MeshDestroy
+
+      ! Dummy argument
+      type(ESMF_Mesh), intent(in) :: new_phys_mesh
+
+      ! Ignore return code here as all we need is an attempt to reclaim memory
+      if (ESMF_MeshIsCreated(phys_mesh)) then
+         call ESMF_MeshDestroy(phys_mesh)
+      end if
+
+      phys_mesh = new_phys_mesh
+
+   end subroutine edyn_esmf_update_phys_mesh
+
```

Comment:
I believe we only need to create the phys_mesh once per run.   The physics grid does not change with time while the magnetic does.  Do we need this update routine?

Reply:
Uh, yeah. This is a workaround to a potential circular dependency issue. All this is doing is setting the mesh in `edyn_esmf`. It is only called from `edyn_gcomp_init` (i.e., only once per run).